improver_tests/utilities/temporal/test_TimezoneExtraction.py

```diff
@@ -0,0 +1,259 @@
+        )
+    return cube
+
+
+def make_timezone_cube():
+    """Makes a timezone cube to use in tests. data=0 where points fall in that
+    time-zone. data=1 where they don't."""
+    cube = set_up_variable_cube(
+        np.zeros((3, 3)).astype(np.float32),
+        name="timezone_mask",
+        units="1",
+        standard_grid_metadata="gl_ens",
+        attributes={
+            "institution": "unknown",
+            "source": "IMPROVER",
+            "title": "Unit test",
+        },
+    )
+    cube = add_coordinate(
+        cube,
+        [-3600, 0, 3600],
+        "UTC_offset",
+        coord_units="seconds",
+        dtype=TIME_COORDS["forecast_period"].dtype,
+    )
+    true_row = [1, 1, 1]
+    false_row = [0, 0, 0]
+    cube.data = np.array(
+        [
+            [true_row, true_row, false_row],
+            [true_row, false_row, true_row],
+            [false_row, true_row, true_row],
+        ],
+        dtype=np.int8,
+    )
+    return cube
+
+
+def assert_metadata_ok(output_cube):
+    """Checks that the meta-data of output_cube are as expected"""
+    assert isinstance(output_cube, Cube)
+    assert output_cube.dtype == np.float32
+    assert output_cube.coord_dims("time") == (0, 1)
+    assert output_cube.coord("time").dtype == np.int64
+    check_mandatory_standards(output_cube)
+
+
+def test_create_output_cube():
+    """Tests that the create_output_cube method builds a cube with appropriate
+    meta-data"""
+    data_shape = [3, 3]
+    cube = make_input_cube(data_shape)
+    local_time = datetime(2017, 11, 9, 12, 0)
+    plugin = TimezoneExtraction()
+    plugin.output_data = np.zeros(data_shape, dtype=np.float32)
+    plugin.time_points = np.full(
+        data_shape,
+        fill_value=Unit(TIME_COORDS["time"].units).date2num(
+            datetime(2017, 11, 10, 4, 0)
+        ),
+        dtype=np.int64,
+    )
+    plugin.time_bounds = None
+    result = plugin.create_output_cube(cube, local_time)
+    assert_metadata_ok(result)
+    assert result.name() == cube.name()
+    assert result.units == cube.units
+    result_local_time = result.coord("time_in_local_timezone")
+    assert [cell.point for cell in result_local_time.cells()] == [local_time]
+    expected_shape = data_shape
+    assert result.shape == tuple(expected_shape)
+    assert result.attributes == cube.attributes
+
+
+@pytest.mark.parametrize(
+    "data_shape, expect_success", (([3, 3], True), ([3, 3, 3], False))
+)
+def test_check_input_cube_dims(data_shape, expect_success):
+    """Checks that check_input_cube_dims can differentiate between an input cube
+    with time, y, x coords and one with time, realization, y, x coords."""
+    cube = make_input_cube(data_shape)
+    plugin = TimezoneExtraction()
+    if expect_success:
+        plugin.check_input_cube_dims(cube)
+    else:
+        with pytest.raises(
+            ValueError, match=r"Expected coords on input_cube: time, y, x "
+        ):
+            plugin.check_input_cube_dims(cube)
```

Comment:
As discussed briefly, this plugin needs to be able to handle an additional dimension (probability or percentile), so the plugin will need to be modified to slice over t,y,x and then reconstruct the other coordinate at the end. As such this function and test will change.

Reply:
I've extended the functionality to allow extra dimensions to be present.